holoviews/core/spaces.py

```diff
@@ -817,6 +817,22 @@ def _cache(self, key, val):
         self.data[key] = val
 
 
+    def map(self, map_fn, specs=None, clone=True):
+        """
+        Recursively replaces elements using a map function when the
+        specification applies. Extends regular map with functionality
+        to dynamically apply functions.
+        """
```

Comment:
Wow! That really was a massive simplification. I like it!

I just hope it works as expected and doesn't become complicated again. Perhaps a few more unit tests would help?